selfdrive/navd/navd.py

```diff
@@ -273,6 +276,11 @@ def send_instruction(self):
     msg.navInstruction.timeRemaining = total_time
     msg.navInstruction.timeRemainingTypical = total_time_typical
 
+    # Compute if we are going the wrong way
+    if self.smallest_distance_remaining is None or total_distance < self.smallest_distance_remaining:
```

Comment:
If you use `math.inf` instead of None you can replace this by a simple ` self.smallest_distance_remaining = min( self.smallest_distance_remaining, total_distance)`